olclient/openlibrary.py

```diff
@@ -135,9 +148,20 @@ def save_many(self, docs, comment):
     @backoff.on_exception(on_giveup=err, **BACKOFF_KWARGS)
```

Comment:
`err = lambda e: logger.exception("Error retrieving OpenLibrary response: %s", e)`
@rahul-kumi This is the exception that the backoff decorator eventually raises/d -- I think we want this to give a clearer idea of what happened, whether it was an 'ID not found', timeout, server error, or something else. I thought the intent of #139 was to avoid raising generic exceptions. The requests exceptions are generally pretty informative, but I think this original code lost some of that information. The messages you have added are good and informative, but I think they need to be encapsulated as custom exceptions (rather than prints) so they can be handled further up the chain.